gobblin-runtime/src/main/java/gobblin/runtime/Task.java

```diff
@@ -279,7 +280,8 @@ private void publishTaskData() throws IOException {
       @SuppressWarnings("unchecked")
       Class<? extends DataPublisher> dataPublisherClass = (Class<? extends DataPublisher>) Class.forName(
           this.taskState.getProp(ConfigurationKeys.DATA_PUBLISHER_TYPE, ConfigurationKeys.DEFAULT_DATA_PUBLISHER_TYPE));
-      DataPublisher publisher = closer.register(DataPublisher.getInstance(dataPublisherClass, this.taskState));
+      SingleTaskDataPublisher publisher =
```

Comment:
Publishing data in tasks is just recently added, unlikely anyone else is using it, even if someone is, chances are that he's extending basedatapublisher rather than directly extending datapublisher, which will be fine.

What's your suggestion?